modules/dnn/src/dnn_common.hpp

```diff
@@ -57,15 +59,17 @@ Net readNetDiagnostic(Args&& ... args)
 class LayerHandler
 {
 public:
-    void addMissing(const std::string& name, const std::string& type);
-    bool contains(const std::string& type) const;
-    void printMissing();
+    void addMissing(const std::string& type, const std::string& name, const std::string& domain = "");
+    bool contains(const std::string& type, const std::string& domain = "") const;
```

Comment:
> domain

My suggestion is to keep domain as part of "type" (for non-default cases) instead of introduction of new entity.

Reply:
The current approach improves readability of the output:

```
[ERROR:0@1.467] global opencv/modules/dnn/src/debug_utils.cpp (86) printMissing DNN: Not supported types:
Type='ScatterElements' (from domain ai.onnx), affected nodes:
['2769', '2744', '2719', '2694']
Type='RoiAlign' (from domain ai.onnx), affected nodes:
['2650', '2635', '2620', '2605']
```

vs

```
[ERROR:0@1.467] global opencv/modules/dnn/src/debug_utils.cpp (86) printMissing DNN: Not supported types:
Type='ai.onnx_ScatterElements' affected nodes:
['2769', '2744', '2719', '2694']
Type='ai.onnx_RoiAlign', affected nodes:
['2650', '2635', '2620', '2605']
```

As a user, I wound be tempted to add `ai.onnx_ScatterElements`, which won't fix the problem in the network and add more useless logs from diagnostic tool about this new layer.

Reply:
`ai.onnx` is default and could be (should be) omitted.

dnn_common.hpp is not about ONNX-only with its domains.